xml.text/test/unit/src/org/netbeans/modules/xml/text/completion/XMLCompletionQueryTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.xml.text.completion;
+
+import java.io.IOException;
+import javax.swing.text.BadLocationException;
+import org.junit.Test;
+import org.netbeans.editor.BaseDocument;
+import org.netbeans.modules.xml.text.AbstractTestCase;
+import org.netbeans.modules.xml.text.api.dom.XMLSyntaxSupport;
+
+public class XMLCompletionQueryTest extends AbstractTestCase {
+
+    public XMLCompletionQueryTest(String testName) {
+        super(testName);
+    }
+
+    @Test
+    public void testShouldCloseTag() throws IOException, BadLocationException, Exception {
+        BaseDocument doc = getDocument("/org/netbeans/modules/xml/text/completion/res/docResourceUnclosed.html");
+        XMLSyntaxSupport xss = XMLSyntaxSupport.getSyntaxSupport((BaseDocument)doc);
+        // The first position is a caret position behind a <a> Element, that
+        // has a closing tag => shouldCloseTag needs to return NULL here to
+        // indicate, that inserted content should be inserted without the
+        // closing tag.
+        SyntaxQueryHelper sqh = new SyntaxQueryHelper(xss, 880);
```

Comment:
Consider to make some markers (weird char sequences) in the document text to indicate positions. The test will be less fragile ;)

Reply:
Good point - I updated the unittest and will merge with the modified test.